components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -233,10 +237,51 @@ public ConnectorConfig getConnectorWithConfigs(String tenantDomain,
 
         for (ConnectorConfig connectorConfig : connectorListWithConfigs) {
             if (connectorConfig.getName().equals(connectorName)) {
+                // Should remove this logic eventually.
+                convertPropertyUseNumericToUseAlphaNumeric(connectorName, connectorConfig);
                 return connectorConfig;
             }
         }
         return null;
     }
 
+    /**
+     * This method is used to make sure property value useNumericCharacters and useAlphanumericCharacters both uses
+     * same user input.
+     *
+     * @param configurationDetails Configuration details of the email OTP connector.
+     */
+    private void convertPropertyUseNumericToUseAlphaNumeric(Map<String, String> configurationDetails) {
+
+        if (configurationDetails.containsKey(EMAIL_OTP_USE_ALPHANUMERIC_CHARS) &&
+                configurationDetails.containsKey(EMAIL_OTP_USE_NUMERIC_CHARS)) {
+            boolean useNumericChars = !Boolean.parseBoolean(configurationDetails.get(EMAIL_OTP_USE_ALPHANUMERIC_CHARS));
+            configurationDetails.put(EMAIL_OTP_USE_NUMERIC_CHARS, String.valueOf(useNumericChars));
+        }
+    }
+
+    /**
+     * This method is used to convert the property value useNumericCharacters to useAlphanumericCharacters.
+     *
+     * @param connectorName   Name of the connector.
+     * @param connectorConfig Connector configuration.
+     */
+    private void convertPropertyUseNumericToUseAlphaNumeric(String connectorName, ConnectorConfig connectorConfig) {
+
+        // If any of the following conditions are false, it simply doesn't migrate the config.
+        if (connectorName.equals(EMAIL_OTP_AUTHENTICATOR) &&
+                connectorConfig.getProperties()[3].getName().equals(EMAIL_OTP_USE_ALPHANUMERIC_CHARS) &&
+                connectorConfig.getProperties()[4].getName().equals(EMAIL_OTP_USE_NUMERIC_CHARS)) {
```

Comment:
how did we get these index values?

Reply:
https://github.com/wso2-extensions/identity-local-auth-emailotp/pull/11/files#r1401745233